graphql/resolvers/alumnis.js

```diff
@@ -118,5 +121,116 @@ module.exports = {
 
       return newAlumni;
     },
+
+    async editAlumniProfile(
+      _,
+      {
+        editAlumniProfileInput: {
+          firstName,
+          lastName,
+          oldEmail,
+          newEmail,
+          undergrad,
+          grad,
+          employer,
+          position,
+          location,
+          linkedin,
+        },
+      }
+    ) {
+      const { errors, valid } = validateEditAlumniProfileInput(
+          firstName,
+          lastName,
+          oldEmail,
+          newEmail,
+          undergrad,
+          grad,
+          employer,
+          position,
+          location,
+          linkedin,
+      );
+
+      var coordinates = {
+        latitude: 0,
+        longitude: 0
+      };
+
+      if (!valid) {
+        handleInputError(errors);
+      }
+
+      grad.year = grad.year === "" ? 0 : grad.year;
+      var email = oldEmail;
+
+      if(newEmail !== oldEmail && newEmail !== ""){
+        const isEmailDuplicate = await Alumni.findOne({ email: newEmail });
+        if (isEmailDuplicate) {
+          errors.general = "that new email already exists.";
+          handleInputError(errors);
+        }
+        email = newEmail;
+      }
+
+      const alumniLocation =
+        location.city +
+        ", " +
+        (location.state ? location.state + ", " : "") +
+        location.country;
+
+      var ngcOptions = {
+        provider: "mapquest",
+        httpAdapter: "https",
+        apiKey: process.env.MQ_KEY,
+        formatter: null,
+      };
+
+      var geocoder = nodegeocoder(ngcOptions);
+
+      await geocoder
+        .geocode(alumniLocation)
+        .then(function (res) {
+          var north = Math.random() * 0.007;
+          var south = -1 * Math.random() * 0.007;
+          var east = Math.random() * 0.007;
+          var west = -1 * Math.random() * 0.007;
+          coordinates.latitude = res[0].latitude + north + south;
+          coordinates.longitude = res[0].longitude + east + west;
+        })
+        .catch(function (err) {
+          errors.general =
+            "Invalid location, please check city, state and/or country.";
+          handleInputError(errors);
+        });
+
+      const alumni = await Alumni.findOne({ email: oldEmail });
+
+      if (alumni) {
+        const updatedAlumni = await Alumni.findOneAndUpdate(
```

Comment:
you don't need to findOne and then findOneAndUpdate. If findOneAndUpdate doesn't find a matching doc it will just return a null.

Additionally I think this query can use the .lean option?

Reply:
Got rid of the first query and used the .lean() option on findOneAndReplace().